src/components/shared/warnPunishedDelegate/index.js

```diff
@@ -49,15 +49,17 @@ const Warning = ({ vote, ...props }) => {
 };
 
 export const DelegateProfileWarning = ({ daysLeft, punishmentStartDate, ...props }) => (
-  <WarnPunishedDelegate
-    daysLeft={daysLeft}
-    punishmentStartDate={punishmentStartDate}
-    {...props}
-  />
+  daysLeft >= 1 ? (
```

Comment:
Why mount the component in the first place? If the punishment is already over, the warning component must not be mounted. I think this condition must be moved to the `editVote` and `wallet/overview` components.